pkg/apis/serving/v1alpha1/servingruntime_types.go

```diff
@@ -0,0 +1,200 @@
+/*
+Copyright 2021 The KServe Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type Framework struct {
+	// +required
+	Name string `json:"name"`
+	// +optional
+	Version *string `json:"version,omitempty"`
+}
```

Comment:
I was wondering whether it would make sense to include some kind of preference ordinal in this struct. To use when there are multiple runtimes satisfying a predictor's model type requirement. _Or_ maybe just a disabled flag here... so that it can be "turned off" for non-preferred runtimes without having to remove from the list.

Reply:
WDYT about just a flag to disable? I'd guess the normal case would be where it's known which the preferred runtime is for a given kind of model, so maybe better to be explicit rather than based on an indirect weight comparison?

Reply:
Hmm, yea, I see what you mean and agree that it is better to be explicit. I am okay with just a disabled flag . As you said, it'd be useful for turning off a model type without removing it from the supported list (which would indicate that the runtime doesn't support a framework/version).

Reply:
@pvaneck I think `framework` is ok as I can see a few references using the same wording e.g
https://azure.microsoft.com/en-us/overview/ml-frameworks.